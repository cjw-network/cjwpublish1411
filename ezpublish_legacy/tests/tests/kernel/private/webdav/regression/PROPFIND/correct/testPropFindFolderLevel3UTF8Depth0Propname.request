<?php

// Test if PROPFIND returns correct information for the directory $GLOBALS['ezc_webdav_testfolder']
// containing a file with UTF-8 characters in the name
// with HTTP_DEPTH = 0
// with propname request instead of prop request

// First upload the file in the folder $GLOBALS['ezc_webdav_testfolder']
$article = new ezpObject( 'file', (int)$GLOBALS['ezc_webdav_testfolderid'] );
$article->name = "româneşte";
$article->file = dirname( __FILE__ ) . DIRECTORY_SEPARATOR . $GLOBALS['ezc_webdav_testfolder'] . '.txt';
$article->publish();

$_SERVER['REQUEST_URI'] = '/' . $GLOBALS['ezc_siteaccess'] . '/Content/' . $GLOBALS['ezc_webdav_testfolder'] . '/';
$_SERVER['REQUEST_METHOD'] = 'PROPFIND';

$_SERVER['HTTP_DEPTH'] = '0';
$_SERVER['CONTENT_TYPE'] = 'application/xml';
$_SERVER['HTTP_CONTENT_LENGTH'] = '288';

// The PROPFIND request needs an XML body
$GLOBALS['ezc_post_body'] = <<<END
<?xml version="1.0" encoding="utf-8"?>
<propfind xmlns="DAV:">
  <propname/>
</propfind>
END;

?>
