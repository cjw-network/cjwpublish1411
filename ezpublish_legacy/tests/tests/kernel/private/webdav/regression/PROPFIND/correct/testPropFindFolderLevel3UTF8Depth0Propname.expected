Content-Type: text/xml; charset="utf-8"
Server: eZComponents/dev/ezcWebdavTransportWrapper
HTTP/1.1 207 Multi-Status

<?xml version="1.0" encoding="UTF-8"?>
<D:multistatus xmlns:D="DAV:">
  <D:response>
    <D:href>http://@ezc_webdav_host@/@ezc_siteaccess@/Content/@ezc_webdav_testfolder@/</D:href>
    <D:propstat>
      <D:prop>
        <D:getcontentlength/>
        <D:getlastmodified/>
        <D:creationdate/>
        <D:displayname/>
        <D:getetag/>
        <D:getcontenttype/>
        <D:resourcetype/>
      </D:prop>
      <D:status>HTTP/1.1 200 OK</D:status>
    </D:propstat>
  </D:response>
</D:multistatus>
